### PR TITLE
New metadata injection implementation

### PR DESCRIPTION
**Pull request by AlexanderBuloichik**

### CODE REVIEW

**Comment by deinspanjer** [comment]
Is there a master issue covering the annotation based MDI support?  Any reason not to squash the base implementation commits?

**Comment by mdamour1976** [comment]
Possibly BACKLOG-5565 for the "master issue."  Regardless, all commits need a JIRA case otherwise there will be violations in an engops report.

**Comment by AlexanderBuloichik** [comment]
@mdamour1976 @deinspanjer Fixed and rebased

**Comment by deinspanjer** [comment]
The new getAsJavaType method looks good, but we need some unit tests to support it.

Once those are included, Mike and I will run the test and coverage report stuff and it should be good to merge.

**Comment by AlexanderBuloichik** [comment]
@deinspanjer I added unit test

**Comment by mdamour1976** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local
3. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local
4. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f ui/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  core/src/org/pentaho/di/core/RowMetaAndData.java
  core/test-src/org/pentaho/di/core/RowMetaAndDataTest.java
  engine/src/org/pentaho/di/core/injection/Injection.java
  engine/src/org/pentaho/di/core/injection/InjectionDeep.java
  engine/src/org/pentaho/di/core/injection/InjectionSupported.java
  engine/src/org/pentaho/di/core/injection/bean/BeanInjectionInfo.java
  engine/src/org/pentaho/di/core/injection/bean/BeanInjector.java
  engine/src/org/pentaho/di/core/injection/bean/BeanLevelInfo.java
  engine/src/org/pentaho/di/trans/TransPainter.java
  engine/src/org/pentaho/di/trans/step/StepMetaInjectionInterface.java
  engine/src/org/pentaho/di/trans/step/StepMetaInterface.java
  engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputField.java
  engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputStepMeta.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileFilter.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMeta.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjection.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/messages/messages_en_US.properties
  engine/src/org/pentaho/di/trans/steps/metainject/MetaInject.java
  engine/src/org/pentaho/di/trans/steps/metainject/MetaInjectData.java
  engine/src/org/pentaho/di/trans/steps/streamlookup/StreamLookupMeta.java
  engine/test-src/org/pentaho/di/core/injection/BaseMetadatInjectionTest.java
  engine/test-src/org/pentaho/di/core/injection/MetaAnnotationInjectionTest.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel1.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel2.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel2Base.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel3.java
  engine/test-src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjectionTest.java
  engine/test-src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaNewInjectionTest.java
  ui/src/org/pentaho/di/ui/trans/steps/metainject/MetaInjectDialog.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File engine/src/org/pentaho/di/trans/TransPainter.java
1. Message: Unused import - org.pentaho.di.trans.step.StepMetaInjectionInterface. at Line: 52 Column: 8

File engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputField.java
1. Message: '{' at column 5 should be on the previous line. at Line: 570 Column: 5

File engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputStepMeta.java
1. Message: '(' is not followed by whitespace. at Line: 130 Column: 16
2. Message: ')' is not preceded with whitespace. at Line: 130 Column: 36
3. Message: '=' is not followed by whitespace. at Line: 130 Column: 21
4. Message: '=' is not preceded with whitespace. at Line: 130 Column: 20
5. Message: '(' is not followed by whitespace. at Line: 134 Column: 16
6. Message: ')' is not preceded with whitespace. at Line: 134 Column: 39
7. Message: '=' is not followed by whitespace. at Line: 134 Column: 21
8. Message: '=' is not preceded with whitespace. at Line: 134 Column: 20
9. Message: '(' is not followed by whitespace. at Line: 138 Column: 16
10. Message: ')' is not preceded with whitespace. at Line: 138 Column: 47
11. Message: '=' is not followed by whitespace. at Line: 138 Column: 21
12. Message: '=' is not preceded with whitespace. at Line: 138 Column: 20
13. Message: '(' is not followed by whitespace. at Line: 141 Column: 16
14. Message: ')' is not preceded with whitespace. at Line: 141 Column: 37
15. Message: '=' is not followed by whitespace. at Line: 141 Column: 21
16. Message: '=' is not preceded with whitespace. at Line: 141 Column: 20
17. Message: '(' is not followed by whitespace. at Line: 145 Column: 16
18. Message: ')' is not preceded with whitespace. at Line: 145 Column: 47
19. Message: '=' is not followed by whitespace. at Line: 145 Column: 21
20. Message: '=' is not preceded with whitespace. at Line: 145 Column: 20
21. Message: '(' is not followed by whitespace. at Line: 149 Column: 16
22. Message: ')' is not preceded with whitespace. at Line: 149 Column: 46
23. Message: '=' is not followed by whitespace. at Line: 149 Column: 21
24. Message: '=' is not preceded with whitespace. at Line: 149 Column: 20
25. Message: '(' is not followed by whitespace. at Line: 153 Column: 16
26. Message: ')' is not preceded with whitespace. at Line: 153 Column: 45
27. Message: '=' is not followed by whitespace. at Line: 153 Column: 21
28. Message: '=' is not preceded with whitespace. at Line: 153 Column: 20
29. Message: '(' is not followed by whitespace. at Line: 157 Column: 16
30. Message: ')' is not preceded with whitespace. at Line: 157 Column: 44
31. Message: '=' is not followed by whitespace. at Line: 157 Column: 21
32. Message: '=' is not preceded with whitespace. at Line: 157 Column: 20
33. Message: '(' is not followed by whitespace. at Line: 161 Column: 16
34. Message: ')' is not preceded with whitespace. at Line: 161 Column: 47
35. Message: '=' is not followed by whitespace. at Line: 161 Column: 21
36. Message: '=' is not preceded with whitespace. at Line: 161 Column: 20
37. Message: '(' is not followed by whitespace. at Line: 165 Column: 16
38. Message: ')' is not preceded with whitespace. at Line: 165 Column: 46
39. Message: '=' is not followed by whitespace. at Line: 165 Column: 21
40. Message: '=' is not preceded with whitespace. at Line: 165 Column: 20
41. Message: 'public' modifier out of order with the JLS suggestions. at Line: 312 Column: 12
42. Message: '=' is not followed by whitespace. at Line: 72 Column: 42
43. Message: '=' is not preceded with whitespace. at Line: 72 Column: 41
44. Message: '=' is not followed by whitespace. at Line: 76 Column: 42
45. Message: '=' is not preceded with whitespace. at Line: 76 Column: 41
46. Message: '=' is not followed by whitespace. at Line: 80 Column: 50
47. Message: '=' is not preceded with whitespace. at Line: 80 Column: 49
48. Message: '=' is not followed by whitespace. at Line: 84 Column: 47
49. Message: '=' is not preceded with whitespace. at Line: 84 Column: 46
50. Message: '=' is not followed by whitespace. at Line: 88 Column: 52
51. Message: '=' is not preceded with whitespace. at Line: 88 Column: 51

File engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMeta.java
1. Message: '(' is not followed by whitespace. at Line: 203 Column: 14
2. Message: ')' is not preceded with whitespace. at Line: 203 Column: 38
3. Message: '=' is not followed by whitespace. at Line: 203 Column: 19
4. Message: '=' is not preceded with whitespace. at Line: 203 Column: 18
5. Message: '(' is not followed by whitespace. at Line: 207 Column: 14
6. Message: ')' is not preceded with whitespace. at Line: 207 Column: 39
7. Message: '=' is not followed by whitespace. at Line: 207 Column: 19
8. Message: '=' is not preceded with whitespace. at Line: 207 Column: 18
9. Message: '(' is not followed by whitespace. at Line: 211 Column: 14
10. Message: ')' is not preceded with whitespace. at Line: 211 Column: 37
11. Message: '=' is not followed by whitespace. at Line: 211 Column: 19
12. Message: '=' is not preceded with whitespace. at Line: 211 Column: 18
13. Message: '(' is not followed by whitespace. at Line: 215 Column: 14
14. Message: ')' is not preceded with whitespace. at Line: 215 Column: 40
15. Message: '=' is not followed by whitespace. at Line: 215 Column: 19
16. Message: '=' is not preceded with whitespace. at Line: 215 Column: 18
17. Message: Unused import - org.pentaho.di.trans.step.StepMetaInjectionInterface. at Line: 61 Column: 8

File engine/src/org/pentaho/di/trans/steps/metainject/MetaInject.java
1. Message: '(' is not followed by whitespace. at Line: 111 Column: 11
2. Message: ')' is not preceded with whitespace. at Line: 111 Column: 59
3. Message: '(' is not followed by whitespace. at Line: 112 Column: 20
4. Message: ')' is not preceded with whitespace. at Line: 112 Column: 30
5. Message: 'if' child have incorrect indentation level 6, expected level should be 8. at Line: 112 Column: None
6. Message: 'method call' child have incorrect indentation level 6, expected level should be 8. at Line: 112 Column: None

File engine/test-src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaNewInjectionTest.java
1. Message: Unused import - org.pentaho.di.core.row.value.ValueMetaFactory. at Line: 31 Column: 8

File ui/src/org/pentaho/di/ui/trans/steps/metainject/MetaInjectDialog.java
1. Message: Line has trailing spaces. at Line: 1227 Column: None
2. Message: '(' is not followed by whitespace. at Line: 1228 Column: 41
3. Message: ')' is not preceded with whitespace. at Line: 1228 Column: 115
4. Message: Line has trailing spaces. at Line: 1280 Column: None
5. Message: '(' is not followed by whitespace. at Line: 1281 Column: 41
6. Message: Line has trailing spaces. at Line: 1323 Column: None



Newly Broken Tests:
-----------------
org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps: junit.framework.AssertionFailedError
```
junit.framework.AssertionFailedError: 
Wanted but not invoked:
metaInject.setEntryValueIfFieldExists(
    refEq({NAME:-(FIRST_NAME)}),
    <any>,
    <any>
);
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:137)

However, there were other interactions with this mock:
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:130)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:130)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:130)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:130)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:130)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:130)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:130)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)
-> at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:131)

	at org.pentaho.di.trans.steps.metainject.MetaInjectTest.injectMetaFromMultipleInputSteps(MetaInjectTest.java:137)

```
org.pentaho.di.core.injection.BaseMetadatInjectionTest.initializationError: java.lang.Exception
```
java.lang.Exception: No runnable methods
	at java.lang.reflect.Constructor.newInstance(Constructor.java:526)

```





Unit test coverage change:
============
org.pentaho.di.core.row.value.ValueMetaBase:
 * instructionPercentCoverageChange : +0.04%
 * branchPercentCoverageChange : +0.16%
 * linePercentCoverageChange : +0.11%
 * complexityPercentCoverageChange : +0.20%
    
org.pentaho.di.core.RowMetaAndData:
 * instructionPercentCoverageChange : +23.63%
 * branchPercentCoverageChange : +57.75%
 * linePercentCoverageChange : +20.58%
 * complexityPercentCoverageChange : +30.00%
 * methodPercentCoverageChange : +2.30%
    
org.pentaho.di.core.row.RowMeta:
 * instructionPercentCoverageChange : +1.17%
 * branchPercentCoverageChange : +1.43%
 * linePercentCoverageChange : +1.49%
 * complexityPercentCoverageChange : +1.54%
 * methodPercentCoverageChange : +3.33%
    
org.pentaho.di.trans.steps.metainject.TargetStepAttribute:
 * instructionPercentCoverageChange :**-47.14%**
 * linePercentCoverageChange :**-33.33%**
 * complexityPercentCoverageChange :**-30.77%**
 * methodPercentCoverageChange :**-44.44%**
    
org.pentaho.di.trans.step.StepInjectionMetaEntry:
 * branchPercentCoverage :50.00%
 * instructionPercentCoverageChange : +24.07%
 * linePercentCoverageChange :**-7.14%**
 * complexityPercentCoverageChange :**-6.67%**
 * methodPercentCoverageChange :**-7.14%**
    
org.pentaho.di.trans.steps.fileinput.text.TextFileFilter:
 * instructionPercentCoverageChange : +5.56%
 * linePercentCoverageChange : +8.33%
 * complexityPercentCoverageChange : +9.09%
 * methodPercentCoverageChange : +9.09%
    
org.pentaho.di.trans.steps.metainject.MetaInject:
 * instructionPercentCoverageChange :**-31.78%**
 * branchPercentCoverageChange :**-28.77%**
 * linePercentCoverageChange :**-35.42%**
 * complexityPercentCoverageChange :**-23.53%**
 * methodPercentCoverageChange :**-51.05%**
    
org.pentaho.di.core.injection.bean.BeanInjectionInfo.Group:
 * linePercentCoverage :80.00%
 * methodPercentCoverage :66.67%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :38.71%
    
org.pentaho.di.trans.steps.textfileinput.TextFileInputMeta:
 * branchPercentCoverageChange :**-0.53%**
    
org.pentaho.di.trans.steps.fileinput.BaseFileInputField:
 * instructionPercentCoverageChange : +1.26%
 * linePercentCoverageChange : +1.89%
 * complexityPercentCoverageChange : +1.37%
 * methodPercentCoverageChange : +2.50%
    
org.pentaho.di.core.injection.bean.BeanInjectionInfo.Property:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :85.71%
 * methodPercentCoverage :50.00%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :63.33%
    
org.pentaho.di.core.injection.bean.BeanInjectionInfo:
 * branchPercentCoverage :60.00%
 * linePercentCoverage :78.79%
 * methodPercentCoverage :85.71%
 * complexityPercentCoverage :58.33%
 * instructionPercentCoverage :66.00%
    
org.pentaho.di.trans.steps.metainject.SourceStepField:
 * instructionPercentCoverageChange :**-5.00%**
 * linePercentCoverageChange :**-5.88%**
 * complexityPercentCoverageChange :**-9.09%**
 * methodPercentCoverageChange :**-14.29%**
    
org.pentaho.di.core.injection.bean.BeanLevelInfo:
 * branchPercentCoverage :72.00%
 * linePercentCoverage :83.61%
 * methodPercentCoverage :87.50%
 * complexityPercentCoverage :63.64%
 * instructionPercentCoverage :77.81%
    
org.pentaho.di.trans.steps.fileinput.text.TextFileInputMeta:
 * instructionPercentCoverageChange :**-0.19%**
 * linePercentCoverageChange :**-0.52%**
 * complexityPercentCoverageChange :**-2.17%**
 * methodPercentCoverageChange :**-5.37%**
    
org.pentaho.di.core.auth.core.impl.DefaultAuthenticationConsumerFactory:
 * branchPercentCoverageChange : +7.14%
 * complexityPercentCoverageChange : +7.69%
    
org.pentaho.di.core.injection.bean.BeanInjector:
 * branchPercentCoverage :55.36%
 * linePercentCoverage :62.77%
 * methodPercentCoverage :80.00%
 * complexityPercentCoverage :51.52%
 * instructionPercentCoverage :51.65%

**Comment by mdamour1976** [comment]
Build Commands:
==============
Build:
------
1. rm -r ~/.kettle; rm -r ~/.pentaho; echo removed hidden folders
2. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local
3. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local
4. ant -Dpentaho.coding.standards.version=1.0.1 -Djunit.jvmargs=-XX:PermSize=256m -f ui/build.xml clean-all resolve jacoco jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove

Changed files:
==============
  core/src/org/pentaho/di/core/RowMetaAndData.java
  core/test-src/org/pentaho/di/core/RowMetaAndDataTest.java
  engine/src/org/pentaho/di/core/injection/Injection.java
  engine/src/org/pentaho/di/core/injection/InjectionDeep.java
  engine/src/org/pentaho/di/core/injection/InjectionSupported.java
  engine/src/org/pentaho/di/core/injection/bean/BeanInjectionInfo.java
  engine/src/org/pentaho/di/core/injection/bean/BeanInjector.java
  engine/src/org/pentaho/di/core/injection/bean/BeanLevelInfo.java
  engine/src/org/pentaho/di/trans/TransPainter.java
  engine/src/org/pentaho/di/trans/step/StepMetaInjectionInterface.java
  engine/src/org/pentaho/di/trans/step/StepMetaInterface.java
  engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputField.java
  engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputStepMeta.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileFilter.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMeta.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjection.java
  engine/src/org/pentaho/di/trans/steps/fileinput/text/messages/messages_en_US.properties
  engine/src/org/pentaho/di/trans/steps/metainject/MetaInject.java
  engine/src/org/pentaho/di/trans/steps/metainject/MetaInjectData.java
  engine/src/org/pentaho/di/trans/steps/streamlookup/StreamLookupMeta.java
  engine/test-src/org/pentaho/di/core/injection/BaseMetadataInjectionTest.java
  engine/test-src/org/pentaho/di/core/injection/MetaAnnotationInjectionTest.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel1.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel2.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel2Base.java
  engine/test-src/org/pentaho/di/core/injection/MetaBeanLevel3.java
  engine/test-src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaInjectionTest.java
  engine/test-src/org/pentaho/di/trans/steps/fileinput/text/TextFileInputMetaNewInjectionTest.java
  ui/src/org/pentaho/di/ui/trans/steps/metainject/MetaInjectDialog.java


Checkstyle violations:
======================
See: https://github.com/pentaho/pentaho-coding-standards

File engine/src/org/pentaho/di/trans/steps/fileinput/BaseFileInputStepMeta.java
1. Message: 'public' modifier out of order with the JLS suggestions. at Line: 312 Column: 12

File ui/src/org/pentaho/di/ui/trans/steps/metainject/MetaInjectDialog.java
1. Message: '(' is not followed by whitespace. at Line: 1228 Column: 41
2. Message: ')' is not preceded with whitespace. at Line: 1228 Column: 115
3. Message: Line has trailing spaces. at Line: 1323 Column: None









Unit test coverage change:
============
org.pentaho.di.core.row.value.ValueMetaBase:
 * instructionPercentCoverageChange : +0.04%
 * branchPercentCoverageChange : +0.16%
 * linePercentCoverageChange : +0.11%
 * complexityPercentCoverageChange : +0.20%
    
org.pentaho.di.core.RowMetaAndData:
 * instructionPercentCoverageChange : +23.63%
 * branchPercentCoverageChange : +57.75%
 * linePercentCoverageChange : +20.58%
 * complexityPercentCoverageChange : +30.00%
 * methodPercentCoverageChange : +2.30%
    
org.pentaho.di.core.row.RowMeta:
 * instructionPercentCoverageChange : +1.17%
 * branchPercentCoverageChange : +1.43%
 * linePercentCoverageChange : +1.49%
 * complexityPercentCoverageChange : +1.54%
 * methodPercentCoverageChange : +3.33%
    
org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta:
 * instructionPercentCoverageChange :**-0.08%**
    
org.pentaho.di.trans.steps.fileinput.text.TextFileFilter:
 * instructionPercentCoverageChange : +5.56%
 * linePercentCoverageChange : +8.33%
 * complexityPercentCoverageChange : +9.09%
 * methodPercentCoverageChange : +9.09%
    
org.pentaho.di.trans.steps.metainject.MetaInject:
 * instructionPercentCoverageChange :**-2.69%**
 * branchPercentCoverageChange :**-2.41%**
 * linePercentCoverageChange :**-2.39%**
 * complexityPercentCoverageChange :**-2.28%**
 * methodPercentCoverageChange :**-4.90%**
    
org.pentaho.di.core.injection.bean.BeanInjectionInfo.Group:
 * linePercentCoverage :80.00%
 * methodPercentCoverage :66.67%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :38.71%
    
org.pentaho.di.job.entries.sql.JobEntrySQL:
 * instructionPercentCoverageChange :**-1.64%**
 * linePercentCoverageChange :**-1.97%**
    
org.pentaho.di.trans.steps.fileinput.BaseFileInputField:
 * instructionPercentCoverageChange : +1.26%
 * linePercentCoverageChange : +1.89%
 * complexityPercentCoverageChange : +1.37%
 * methodPercentCoverageChange : +2.50%
    
org.pentaho.di.trans.steps.replacestring.ReplaceStringMeta:
 * instructionPercentCoverageChange : +0.07%
    
org.pentaho.di.core.injection.bean.BeanInjectionInfo.Property:
 * branchPercentCoverage :100.00%
 * linePercentCoverage :85.71%
 * methodPercentCoverage :50.00%
 * complexityPercentCoverage :66.67%
 * instructionPercentCoverage :63.33%
    
org.pentaho.di.core.injection.bean.BeanInjectionInfo:
 * branchPercentCoverage :60.00%
 * linePercentCoverage :78.79%
 * methodPercentCoverage :85.71%
 * complexityPercentCoverage :58.33%
 * instructionPercentCoverage :66.00%
    
org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * instructionPercentCoverageChange :**-0.17%**
    
org.pentaho.di.core.injection.bean.BeanLevelInfo:
 * branchPercentCoverage :72.00%
 * linePercentCoverage :83.61%
 * methodPercentCoverage :87.50%
 * complexityPercentCoverage :63.64%
 * instructionPercentCoverage :77.81%
    
org.pentaho.di.core.lifecycle.KettleLifecycleSupport:
 * branchPercentCoverageChange :**-8.33%**
 * complexityPercentCoverageChange :**-9.09%**
    
org.pentaho.di.trans.steps.fileinput.text.TextFileInputMeta:
 * instructionPercentCoverageChange :**-0.19%**
 * linePercentCoverageChange :**-0.52%**
 * complexityPercentCoverageChange :**-2.17%**
 * methodPercentCoverageChange :**-5.37%**
    
org.pentaho.di.core.auth.core.impl.DefaultAuthenticationConsumerFactory:
 * branchPercentCoverageChange :**-7.14%**
 * complexityPercentCoverageChange :**-7.69%**
    
org.pentaho.di.core.injection.bean.BeanInjector:
 * branchPercentCoverage :55.36%
 * linePercentCoverage :62.77%
 * methodPercentCoverage :80.00%
 * complexityPercentCoverage :51.52%
 * instructionPercentCoverage :51.65%

File: ui/src/org/pentaho/di/ui/trans/steps/metainject/MetaInjectDialog.java [hunk 1]
```diff
@@ -1266,6 +1225,103 @@ public void handleEvent( Event arg0 ) {
     }
   }
 
+  private void processOldMDIDescription(StepMeta stepMeta, TreeItem stepItem, StepMetaInjectionInterface injection) throws KettleException {
```
Comment (mdamour1976) [inline ui/src/org/pentaho/di/ui/trans/steps/metainject/MetaInjectDialog.java]: @AlexanderBuloichik yeah this line has some checkstyle problems

**Pull request merged**